rtic-sync/src/channel.rs

```diff
@@ -497,11 +559,18 @@ impl<T, const N: usize> Receiver<'_, T, N> {
     /// Receives a value if there is one in the channel, non-blocking.
     pub fn try_recv(&mut self) -> Result<T, ReceiveError> {
         // Try to get a ready slot.
-        let ready_slot = critical_section::with(|cs| self.0.access(cs).readyq.pop_front());
+        let ready_slot = critical_section::with(|cs| unsafe {
+            // SAFETY: `self.0.readyq` is not called recursively.
+            self.0.readyq(cs, |q| q.pop_front())
+        });
 
         if let Some(rs) = ready_slot {
             // Read the value from the slots, note; this memcpy is not under a critical section.
-            let r = unsafe { ptr::read(self.0.slots.get_unchecked(rs as usize).get() as *const T) };
+            let r = unsafe {
+                let first_element = self.0.slots.get_unchecked(rs as usize).get_mut();
+                let ptr = first_element.deref().as_ptr();
```

Comment:
Same comment as earlier, why s this now a large block?

Reply:
the extra `get_mut()`, `deref()` and `as_ptr()` (all required to fully get the same thing we had before) would've expanded the single line to more than one anyways. I figured not cramming them into the `ptr::read` was more legible, but am open to better suggestions.

Reply:
Alright, due to the interface it's what we get.